### PR TITLE
[CABMAN] Replace linked lists with std::list

### PR DESCRIPTION
**Pull request by learn-more**
JIRA issue: [CORE-17231](https://jira.reactos.org/browse/CORE-17231)
In preparation of  [CORE-17258](https://jira.reactos.org/browse/CORE-17258)

### CODE REVIEW

File: sdk/tools/cabman/cabinet.cxx [hunk 1]
```diff
@@ -272,22 +257,11 @@ void CCabinet::DestroySearchCriteria()
  * FUNCTION: Destroys the list with the search criteria
  */
 {
-    PSEARCH_CRITERIA Criteria;
-    PSEARCH_CRITERIA NextCriteria;
-
-    Criteria = CriteriaListHead;
-
-    while(Criteria)
+    for (PSEARCH_CRITERIA Criteria : CriteriaList)
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:260]: grmbl

File: sdk/tools/cabman/cabinet.cxx [hunk 2]
```diff
@@ -537,16 +510,14 @@ ULONG CCabinet::Open()
         }
 
         /* Read data blocks for all folders */
-        FolderNode = FolderListHead;
-        while (FolderNode != NULL)
+        for (PCFFOLDER_NODE Node : FolderList)
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:513]: grmbl 2

File: sdk/tools/cabman/cabinet.cxx [hunk 3]
```diff
@@ -591,54 +562,49 @@ ULONG CCabinet::FindNext(PCAB_SEARCH Search)
...
-        Search->Next  = FileListHead;
+        Search->Next  = FileList.begin();
 
         /* Skip split files already extracted */
-        while ((Search->Next) &&
-            (Search->Next->File.FileControlID > CAB_FILE_MAX_FOLDER) &&
-            (Search->Next->File.FileOffset <= LastFileOffset))
+        while ((Search->Next != FileList.end()) &&
+            ((*Search->Next)->File.FileControlID > CAB_FILE_MAX_FOLDER) &&
+            ((*Search->Next)->File.FileOffset <= LastFileOffset))
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:574]: indentation seems broken.

File: sdk/tools/cabman/cabinet.cxx [hunk 4]
```diff
@@ -591,54 +562,49 @@ ULONG CCabinet::FindNext(PCAB_SEARCH Search)
...
+                (*Search->Next)->FileName.c_str(), (UINT)(*Search->Next)->File.FileOffset, (UINT)LastFileOffset));
+            Search->Next++;
         }
 
         RestartSearch = false;
     }
 
     /* Check each search criteria against each file */
-    while(Search->Next)
+    while(Search->Next != FileList.end())
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:585]:
```suggestion
    while (Search->Next != FileList.end())
```

File: sdk/tools/cabman/cabinet.cxx [hunk 5]
```diff
@@ -591,54 +562,49 @@ ULONG CCabinet::FindNext(PCAB_SEARCH Search)
...
         // Some features (like displaying cabinets) don't require search criteria, so we can just break here.
         // If a feature requires it, handle this in the ParseCmdline() function in "main.cxx".
-        if(!CriteriaListHead)
+        if (CriteriaList.empty())
             break;
 
-        Criteria = CriteriaListHead;
-
-        while(Criteria)
+        for (PSEARCH_CRITERIA Criteria : CriteriaList)
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:592]: grmbl 3

File: sdk/tools/cabman/cabinet.cxx [hunk 6]
```diff
@@ -1463,8 +1424,7 @@ ULONG CCabinet::CommitDisk(ULONG MoreDisks)
     WriteFileEntries();
 
     /* Write data blocks */
-    FolderNode = FolderListHead;
-    while (FolderNode != NULL)
+    for (PCFFOLDER_NODE FolderNode : FolderList)
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:1427]: grmbl 4

File: sdk/tools/cabman/cabinet.cxx [hunk 7]
```diff
@@ -1634,9 +1592,7 @@ bool CCabinet::CreateSimpleCabinet()
     }
 
     // Add each file in the criteria list
-    Criteria = CriteriaListHead;
-
-    while(Criteria)
+    for (PSEARCH_CRITERIA Criteria : CriteriaList)
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:1595]: grmbl 5

File: sdk/tools/cabman/cabinet.cxx [hunk 8]
```diff
@@ -1862,24 +1816,20 @@ PCFFOLDER_NODE CCabinet::LocateFolderNode(ULONG Index)
...
 
         case CAB_FILE_CONTINUED:
         case CAB_FILE_PREV_NEXT:
-            return FolderListHead;
+            return FolderList.front();
     }
 
-    Node = FolderListHead;
-    while (Node != NULL)
+    for (PCFFOLDER_NODE Node : FolderList)
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:1829]: grmbl 6

File: sdk/tools/cabman/cabinet.cxx [hunk 9]
```diff
@@ -1894,15 +1844,12 @@ ULONG CCabinet::GetAbsoluteOffset(PCFFILE_NODE File)
...
-    PCFDATA_NODE Node;
-
     DPRINT(MAX_TRACE, ("FileName '%s'  FileOffset (0x%X)  FileSize (%u).\n",
         File->FileName.c_str(),
         (UINT)File->File.FileOffset,
         (UINT)File->File.FileSize));
 
-    Node = CurrentFolderNode->DataListHead;
-    while (Node != NULL)
+    for (PCFDATA_NODE Node : CurrentFolderNode->DataList)
```
Comment (HBelusca) [inline sdk/tools/cabman/cabinet.cxx:1852]: grmbl 7

**Review by HBelusca — COMMENTED** [review]
ok there is some syntax I don't particularly appreciate.

File: sdk/tools/cabman/cabinet.cxx (hunk 4 above)
Comment (learn-more) [reply sdk/tools/cabman/cabinet.cxx:585]: I just updated the code to work with lists, I did not re-format the entire module so I'd prefer to keep formatting changes out of this.

File: sdk/tools/cabman/cabinet.cxx (hunk 3 above)
Comment (learn-more) [reply sdk/tools/cabman/cabinet.cxx:574]: indentation is exactly as it was.

**Pull request merged**